gdsfactory/technology/layer_stack.py

```diff
@@ -170,6 +172,19 @@ def get_component_with_net_layers(
                 else:
                     net_component.add_polygon(polygon, layer=port.layer)
 
+        # Filter the layerstack for layers that are present
+        if remove_empty_layers:
+            layers_present = net_component.layers
+            layernames_dict = self.get_layer_to_layername()
+            layernames_present = [
+                name
+                for sublist in [layernames_dict[layer] for layer in layers_present]
+                for name in sublist
+            ]
+            for key in list(self.layers.keys()):
+                if key not in layernames_present:
+                    del self.layers[key]
```

Comment:
This is pretty dangerous

Reply:
Returning a new LayerStack is better for sure. Since this function is a utility for simulators, there has been no reason so far to run it twice in the same Python session